src/dspbase.jl

```diff
@@ -677,6 +676,9 @@ function _conv(u, v, su, sv)
     _conv!(out, u, v, su, sv, outsize)
 end
 
+# We use this type definition for clarity
+const AbstractFloats = Union{AbstractFloat,Complex{T} where T<:AbstractFloat}
```

Comment:
```suggestion
const AbstractFloats = Union{<:AbstractFloat,Complex{T} where T<:AbstractFloat}
```

Reply:
Can you tell me the difference between the two?

Reply:
hi, when you allow inheritance from the non-concrete type then the type inference system can better resolve potentially down to concrete types.  If you force non-concrete types then the type inference will stop at the non-concrete and require dynamic dispatch at run time.  I was a bit confused by this in the past myself, where the best examples go something like this:
```julia
ff(x::AbstractVector) = # allows inference to concrete types
ff(x::Union{AbstractVector,Real}) = # restricts to non-concrete types
ff(x::Union{<:AbstractVector, <:Real) = # allows inference down to concrete types
```

More pronounced in types (different example):
```julia
struct MyType1
  # restricts to non-concrete type
  dict::Dict{Symbol, AbstractSting}
end

struct MyType2
  # allows type inference try find anonymous concrete type
  dict::Dict{Symbol, <:AbstractSting}
end
```

Maybe newer versions of Julia cleaned up the syntax on this, but I'm just in the habit of adding the `<:` for performance and noticed it on the PR here.

Reply:
Thanks!

Reply:
@dehann could you give a concrete example where adding the `<:` inside a `Union` like here is beneficial? (Note that `Union{<:AbstractFloat,Complex{T} where T<:AbstractFloat} == Union{AbstractFloat,Complex{T} where T<:AbstractFloat}` while `Dict{Symbol, <:AbstractString} != Dict{Symbol, AbstractString}`.) Otherwise I'd be in favor of removing the `<:`.

Reply:
Hmm, these two functions shows that they both allow inference to concrete subtypes of one of the partners in the `Union`.
```julia
julia> ff(x::Union{AbstractVector, Real}) = norm(x)
ff (generic function with 1 method)

julia> @code_warntype ff([1,2,3])
MethodInstance for ff(::Vector{Int64})
  from ff(x::Union{Real, AbstractVector}) in Main at REPL[35]:1
Arguments
  #self#::Core.Const(ff)
  x::Vector{Int64}
Body::Float64
1 ─ %1 = Main.norm(x)::Float64
└──      return %1


julia> gg(x::Union{<:AbstractVector, <:Real}) = norm(x)
gg (generic function with 1 method)

julia> @code_warntype gg([1,2,3])
MethodInstance for gg(::Vector{Int64})
  from gg(x::Union{Real, AbstractVector}) in Main at REPL[38]:1
Arguments
  #self#::Core.Const(gg)
  x::Vector{Int64}
Body::Float64
1 ─ %1 = Main.norm(x)::Float64
└──      return %1
```

Presumably it didn't matter in the case at hand because the type union was always used as a type parameter, which should in and of itself allow inference.

There's also nothing in the documentation about type unions on using `<:` https://docs.julialang.org/en/v1/manual/types/#Type-Unions